app/api/routes/product.py

```diff
@@ -1,148 +1,108 @@
-        }
-        products_list.append(product_dict)
-    return products_list
-    
-""" Search for Product"""
-@router.get("/products/search",status_code=200,response_model=dict)
-async def search_products(product: ProductSearchParams):
-    name = product.name
-    min_price = product.min_price
-    max_price = product.max_price
-    isAvailable = product.isAvailable
-    page = product.page
-    page_size = product.page_size
-    sort_by = product.sort_by
-    sort_order = product.sort_order
-    
-    filtered_products = []
-# Filling the array 
-    for product_id, product_data in products_db.items():
-        if name and name.lower() not in product_data["name"].lower():
-            continue
-        if min_price and product_data["price"] < min_price:
-            continue
-        if max_price and product_data["price"] > max_price:
-            continue
-        if isAvailable is not None and product_data["isAvailable"] != isAvailable:
-            continue
-        filtered_products.append({
-            "id": product_id,
-            "name": product_data["name"],
-            "price": product_data["price"],
-            "stock": product_data["stock"],
-            "isAvailable": product_data["isAvailable"],
-            "created_at": product_data["created_at"]
-        })
-        
-    if not filtered_products:
-        return []
-    
-    if sort_order == "asc":
-        filtered_products.sort(key=lambda x: x[sort_by])
-    elif sort_order == "desc":
-        filtered_products.sort(key=lambda x: x[sort_by], reverse=True)
-    total_products = len(filtered_products)
-    total_pages = (total_products + page_size - 1) // page_size
-    start = (page - 1) * page_size
-    end = start + page_size
-    paginated_products = filtered_products[start:end]
-    
-    return {
-        "page": page,
-        "total_pages": total_pages,
-        "products_per_page": page_size,
-        "total_products": total_products,
-        "products": paginated_products
-    }
\ No newline at end of file
+# Endpoint to list all products
+@router.get("/products", response_model=List[schemas.Product], status_code=status.HTTP_200_OK)
+def list_products_endpoint(
+    db: Session = Depends(database.get_db),
+    name: Optional[str] = None,
+    min_price: Optional[float] = None,
+    max_price: Optional[float] = None,
+    is_available: Optional[bool] = None,
+    page: int = 1,
+    page_size: int = 10,
+    sort_by: str = "created_at",
+    sort_order: str = "asc"
+):
+    products = product_service.list_products(
+        db, name=name, min_price=min_price, max_price=max_price, is_available=is_available, 
+        page=page, page_size=page_size, sort_by=sort_by, sort_order=sort_order
+    )
+    return products
+# Endpoint to search for products based on query params
+@router.get("/products/search", response_model=List[schemas.ProductResponse], status_code=status.HTTP_200_OK)
+def search_products_endpoint(
```

Comment:
You can define a query parameter in a model. Check out [this](https://fastapi.tiangolo.com/tutorial/query-param-models/#query-parameters-with-a-pydantic-model) example.

Reply:
Done

Reply:
I put it in the schemas file

Reply:
Please don't resolve the review conversation yourself. 
In this review, I was referring to the query parameter. Instead of passing these values one by one, you can use a model. Please double-check the link in the first comment.

Reply:
Done